qiskit/providers/models/backendproperties.py

```diff
@@ -122,8 +120,143 @@ def __init__(self, backend_name, backend_version, last_update_date,
+                formatted_props[param.name] = (value, param.date)
+            self._gates[gate.gate][qubits] = formatted_props
+
         super().__init__(**kwargs)
+
+    def gate_error(self, operation: str, qubits: Union[int, Iterable[int]]):
+        """
+        Return gate error estimates from backend properties.
+        Args:
+            operation (str): The operation for which to get the error.
+            qubits (Union[int, Iterable[int]]): The specific qubits for the operation.
+
+        Raises:
+	        PulseError: If error is True and the property is not found.
+        """
+        try:
+            result = self._gates.get(operation).get(_to_tuple(qubits)).get('gate_error')[0]
+            # Throw away datetime at index 1
+            # return self.gates[operation][_to_tuple(qubits)]['gate_error'][0]
+        except (KeyError, TypeError):
+            raise PulseError("Could not find the desired property.")
+        return result
+
+    def gate_length(self, operation: str, qubits: Union[int, Iterable[int]]):
+        """
+        Return the duration of the gate in units of seconds.
+        Args:
+            operation (str): The operation for which to get the duration.
+            qubits (Union[int, Iterable[int]]): The specific qubits for the operation.
+
+        Raises:
+	        PulseError: If error is True and the property is not found.
+        """
+        try:
+            result = self._gates.get(operation).get(_to_tuple(qubits)).get('gate_length')[0]
+            # Throw away datetime at index 1
+            # return self.gates[operation][_to_tuple(qubits)]['gate_length'][0]
+        except KeyError:
+            raise PulseError("Could not find the desired property.")
+        return result
+
+    def get_gate_property(self,
+                          gate: str = None,
+                          qubits: Union[int, Iterable[int]] = None,
+                          gate_property: str = None) -> Tuple[Any, datetime.datetime]:
+        """
+        Return the gate properties of the given qubit and property, if it was given by the
+        backend, otherwise, return `None` or raise an error.
+
+        Args:
+            gate (str) : Name of the gate
+            qubits (Union[int, Iterable[int]]): The property to look for.
+            gate_property (str): Optionally used to specify within the heirarchy which
+            property to return.
+
+        Raises:
+	        PulseError: If error is True and the property is not found.
+        """
+        result = self._gates
+        try:
+            if gate is not None:
+                result = result[gate]
+                if qubits is not None:
+                    result = result[_to_tuple(qubits)]
+                    if gate_property is not None:
+                        result = result[gate_property]
+        except (KeyError, TypeError):
+            raise PulseError("Could not find the desired property.")
```

Comment:
Could we be more verbose or helpful? Once `operation` is a required argument, we could include that in the message

Reply:
I think we should probably catch `KeyError` and `TypeError` separately. They aren't raised for the same reason. If we see a `KeyError`, then whatever the user is looking for just isn't available. For a `TypeError`, there's something else wrong with the request. (I think it means it's over specified... is that something that can still happen? We can write some tests to find out.)